### PR TITLE
Refactoring, fixes to scorm role

### PR DESCRIPTION
**Pull request by bryanlandia**
A lot of this had been done already in the Dogwood configuration branch (refactoring into several playbooks).  This adds a playbook under `appsemblerPlaybooks` to run just the scorm role to make it easier to use from Ax, and makes some fixes to variables and how the SCORM player is checked out.

### CODE REVIEW

**Comment by bryanlandia** [comment]
@OmarIthawi - You won't be familiar with this.  We have an XBlock that supports using SCORM assets.  It relies on a third party commercial "player"  providing HTML and JS to handle most SCORM-specific tasks in the front-end, and then communicates/retrieves a status to the LMS via an XBlock handler and  kicks off grading.  This role does a bunch of necessary setup for getting the player, making a directory for SCORM package asset storage, backup of those assets to S3, and configuration of Nginx.

File: playbooks/edx-east/edxapp.yml [hunk 1]
```diff
@@ -9,7 +9,6 @@
   roles:
    - aws
    - role: scorm
-      when: "{{ EDXAPP_XBLOCK_SETTINGS }}.get('ScormXBlock', False)"
```
Comment (OmarIthawi) [inline playbooks/edx-east/edxapp.yml]: I was going to report this line @bryanlandia! Thanks for fixing the error!

File: playbooks/edx-east/edxapp.yml [hunk 2]
```diff
@@ -29,4 +28,4 @@
    - role: newrelic
      when: COMMON_ENABLE_NEWRELIC
    - role: minos
-     when: COMMON_ENABLE_MINOS
+     when: COMMON_ENABLE_MINOS
```
Comment (OmarIthawi) [inline playbooks/edx-east/edxapp.yml]: Niptuck: looks like we have a noise here. Let's avoid such changes when possible.

File: playbooks/roles/scorm/tasks/scorm_base.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
+
+- name: FIXME create application user (copied over from edxapp/tasks/main.yml)
```
Comment (OmarIthawi) [inline playbooks/roles/scorm/tasks/scorm_base.yml]: It's not clear what the `FIXME` suggests. Could you please either remove it or add more details.

File: playbooks/roles/scorm/templates/sync-scorm-content-to-s3.j2 [hunk 4]
```diff
@@ -62,4 +62,4 @@ onerror() {
 
 trap onerror ERR SIGHUP SIGINT SIGTERM
 
-s3cmd sync -c {{ scorm_dirs.home.path }}/.s3cfg $noop {{ scorm_asset_local_storage_path }}/ "s3://{{ EDXAPP_AWS_STORAGE_BUCKET_NAME }}/scorms/" >>{{scorm_s3_sync_logfile}} 2>&1
+s3cmd sync -c {{ scorm_dirs.home.path }}/.s3cfg $noop {{ scorm_asset_local_storage_path }}/ "s3://{{ EDXAPP_AWS_STORAGE_BUCKET_NAME }}/scorms/" >>{{scorm_s3_sync_logfile}} 2>&1
```
Comment (OmarIthawi) [inline playbooks/roles/scorm/templates/sync-scorm-content-to-s3.j2]: Niptuck: Same noise here as well. (A newline was removed, which is not necessary and clobbers the git log).

File: playbooks/roles/scorm/tasks/main.yml [hunk 5]
```diff
@@ -1,101 +1,22 @@
...
-    path="{{ item.value.path }}"
-    state="directory"
-    owner="{{ item.value.owner }}"
-    group="{{ item.value.group }}"
-    mode="{{ item.value.mode }}"
-  with_dict: scorm_dirs
-  when: EDXAPP_XBLOCK_SETTINGS.ScormXBlock
+- name: define scorm config from settings if available
+  set_fact: 
+    scorm_config: "{{ EDXAPP_XBLOCK_SETTINGS.ScormXBlock|default('{}') }}"
```
Comment (OmarIthawi) [inline playbooks/roles/scorm/tasks/main.yml]: Does this expression work? (I don't know). If it does then keep it.

File: playbooks/roles/scorm/tasks/scorm_players.yml [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+    force=yes owner={{ scorm_player_dir.owner }} mode=0600
+  with_dict: "{{scorm_config.SCORM_PLAYER_BACKENDS|default({})}}"
+
+- name: checkout player repo
+  git: >
+    dest={{ scorm_player_dir.path }}/{{item.key}} repo={{ item.value.repo }} version="{{ item.value.version }}"
+    accept_hostkey=yes key_file=/tmp/scormplayer_{{ item.key }}_identity
+    ssh_opts="-o StrictHostKeyChecking=no"
+  environment:
+    GIT_SSH: "{{ edxapp_git_ssh }}"
```
Comment (OmarIthawi) [inline playbooks/roles/scorm/tasks/scorm_players.yml]: We have both `edxapp_git_ssh` and `key_file` in the same command. I think `edxapp_git_ssh` is useless here.

File: playbooks/roles/scorm/tasks/main.yml (hunk 5 above)
Comment (bryanlandia) [reply playbooks/roles/scorm/tasks/main.yml]: Yes, it works :) .  Just ran it yesterday.

File: playbooks/roles/scorm/tasks/scorm_players.yml (hunk 6 above)
Comment (bryanlandia) [reply playbooks/roles/scorm/tasks/scorm_players.yml]: True, thanks.  I removed it.

File: playbooks/roles/scorm/templates/sync-scorm-content-to-s3.j2 (hunk 4 above)
Comment (bryanlandia) [reply playbooks/roles/scorm/templates/sync-scorm-content-to-s3.j2]: removed.

File: playbooks/roles/scorm/tasks/scorm_base.yml (hunk 3 above)
Comment (bryanlandia) [reply playbooks/roles/scorm/tasks/scorm_base.yml]: I updated with a better comment.

File: playbooks/edx-east/edxapp.yml (hunk 2 above)
Comment (bryanlandia) [reply playbooks/edx-east/edxapp.yml]: removed.

File: playbooks/edx-east/edxapp.yml (hunk 1 above)
Comment (bryanlandia) [reply playbooks/edx-east/edxapp.yml]: 👍

**Pull request merged**